src/context/context.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useReducer } from "react";
+import { filterReducer } from "./reducer";
+
+const Context = createContext();
+
+const Provider = ({ children }) => {
+
+	const [filterState, filterDispatch] = useReducer(filterReducer, {
+		excludeNotAvailable: false,
+		offerItems: false,
+		byRating: "",
+		isSort: "",
```

Comment:
you can put this as a constant and import in reducer also to avoid code duplication